Code/Mantid/Framework/API/src/Projection.cpp

```diff
@@ -38,6 +40,73 @@ Projection::Projection(const V3D &u, const V3D &v, const V3D &w) {
   }
 }
 
+Projection::Projection(ITableWorkspace_const_sptr ws) {
+  if (!ws)
+    throw std::runtime_error(
+        "Null ITableWorkspace given to Projection constructor");
+
+  if (ws->columnCount() != 4)
+    throw std::runtime_error(
+        "4 columns must be provided to create a projection");
+
+  const size_t numRows = ws->rowCount();
+  if (numRows != 3)
+    throw std::runtime_error("3 rows must be provided to create a projection");
+
+  if (ws->getColumn("name")->size() != numRows)
+    throw std::runtime_error("Insufficient values in 'name' column.");
+  if (ws->getColumn("value")->size() != numRows)
+    throw std::runtime_error("Insufficient values in 'value' column.");
+  if (ws->getColumn("offset")->size() != numRows)
+    throw std::runtime_error("Insufficient values in 'offset' column.");
+  if (ws->getColumn("type")->size() != numRows)
+    throw std::runtime_error("Insufficient values in 'unit' column.");
```

Comment:
Ah, yes. I put the checks in because I was getting a segfault. It turned out the segfault was caused by me pulling a `std::string` from a `double` column, which I subsequently fixed, but forgot to remove these checks.

I'll remove them now they're not needed.